src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Credential/CredentialHandler.cs

```diff
@@ -60,6 +63,36 @@ public override Task<string> Script(CredentialViewContext context, CredentialInf
             throw new NotImplementedException();
         }
 
+        public async Task Create(Contracts.CreateCredentialRequestParams parameters)
+        {
+            await ConfigureCredential(parameters.ConnectionUri, parameters.CredentialInfo, ConfigAction.Create, RunType.RunNow);
+        }
+
+        public List<string> GetCredentials(Contracts.GetCredentialsRequestParams parameters)
+        {
+            List<string> credentials = new List<string>();
+            ConnectionInfo connectionInfo = this.GetConnectionInfo(parameters.ConnectionUri);
+            using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo))
+            {
+                if (sqlConn != null)
+                {
+                    using (var cmd = new SqlCommand { Connection = sqlConn })
+                    {
+                        cmd.CommandText = "SELECT [NAME] FROM sys.credentials";
```

Comment:
Since this is just getting the name, maybe name this `getCredentialNames`? Otherwise it sounds like it should be returning the actual secret too